numpy/core/src/multiarray/compiled_base.c

```diff
@@ -315,7 +315,8 @@ arr_insert_loop(char *mptr, char *vptr, char *input_data, char *zero,
                 int totmask, int numvals, int nd, npy_intp *instrides,
                 npy_intp *inshape)
 {
-    int mindx, rem_indx, indx, i, copied;
+    int mindx, rem_indx, i, copied;
```

Comment:
`mindx`, `rem_indx` and `indx` should all be of type `npy_intp`, which is NumPy's version of `ssize_t`.